docs/index.bs

```diff
@@ -193,15 +193,31 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
         * Detects abnormal operation: such as infinite loops and tasks exceeding imposed time limits (if any) while handling the events.
     </section>
 
-    <section>
+   <section>
       <h4 id="service-worker-events">Events</h4>
 
       The Service Workers specification defines <dfn export id="dfn-service-worker-events">service worker events</dfn> (each of which is an [=event=]) that include (see the <a href="#execution-context-events">list</a>):
           * <dfn export id="dfn-lifecycle-events">Lifecycle events</dfn>: {{install!!event}} and {{activate!!event}}.
           * <dfn export id="dfn-functional-events">Functional events</dfn>: {{fetch!!event}} and the [=events=] defined by other specifications that <a href="#extensibility">extend</a> the Service Workers specification. (See the <a href="#execution-context-events">list</a>.)
           * {{message!!event}} and {{messageerror!!event}}.
     </section>
-  </section>
+    <section>
+      <h4 id="service-worker-timing">Timing</h4>
+
+      Service workers mark certain points in times, that are later exposed by the {{PerformanceNavigationTiming|navigation timing}} API.
+
+      A [=/response=] has an associated null or [=service worker timing info=] <dfn export for=response>service worker timing</dfn>, initially set to null.
```

Comment:
I would prefer if we defined this in Fetch and SW just does the marking. Mainly to keep response defined in a single place.

Is it correct that this information gets lost if a network error is returned instead at some point?

Reply:
> I would prefer if we defined this in Fetch and SW just does the marking. Mainly to keep response defined in a single place.

OK. I was trying to make it so that FETCH doesn't have to be modified every time we add a new service-worker metric. Should the struct live in FETCH as well?

> Is it correct that this information gets lost if a network error is returned instead at some point?

This is only exposed through Navigation Timing, if there's a network response the document won't be loaded and these values would be meaningless. In other words, correct.

Reply:
I think it's okay to keep the struct here to make that easier.

Reply:
@jakearchibald I want to make sure we're OK with how this whole process works, and then I'll publish:

1. A SW PR that defines the new struct
2. A FETCH PR that associates the struct with a response and adds arguments to the SW FETCH call
3. The marking part of the SW PR